test/unit/mongo_logger.test.ts

```diff
@@ -0,0 +1,204 @@
+
+    context('when component severities are not set or invalid', function () {
+      it('only uses the default severity for those components', function () {
+        const loggerOptions = MongoLogger.resolveOptions(
+          {
+            MONGODB_LOG_COMMAND: '',
+            MONGODB_LOG_TOPOLOGY: undefined,
+            MONGODB_LOG_SERVER_SELECTION: 'invalid',
+            MONGODB_LOG_CONNECTION: SeverityLevel.EMERGENCY,
+            MONGODB_LOG_ALL: SeverityLevel.CRITICAL
+          },
+          {}
+        );
+        expect(loggerOptions.command).to.equal(loggerOptions.defaultSeverity);
+        expect(loggerOptions.topology).to.equal(loggerOptions.defaultSeverity);
+        expect(loggerOptions.serverSelection).to.equal(loggerOptions.defaultSeverity);
+        expect(loggerOptions.connection).to.equal(SeverityLevel.EMERGENCY);
+      });
+    });
+
+    context('when environment variables and client options are both set', function () {
+      it('gives precedence to environment variables', function () {
+        const loggerOptions = MongoLogger.resolveOptions(
+          { MONGODB_LOG_PATH: 'env' },
+          { mongodbLogPath: 'client' }
+        );
+        expect(loggerOptions.logDestination).to.equal('env');
+      });
+    });
+
+    context('maxDocumentLength', function () {
+      context('when MONGODB_LOG_MAX_DOCUMENT_LENGTH is undefined', function () {
+        it('defaults to 1000', function () {
+          const loggerOptions = MongoLogger.resolveOptions(
+            {
+              MONGODB_LOG_MAX_DOCUMENT_LENGTH: undefined
+            },
+            {}
+          );
+          expect(loggerOptions.maxDocumentLength).to.equal(1000);
+        });
+      });
+
+      context('when MONGODB_LOG_MAX_DOCUMENT_LENGTH is an empty string', function () {
+        it('defaults to 1000', function () {
+          const loggerOptions = MongoLogger.resolveOptions(
+            {
+              MONGODB_LOG_MAX_DOCUMENT_LENGTH: ''
+            },
+            {}
+          );
+          expect(loggerOptions.maxDocumentLength).to.equal(1000);
+        });
+      });
+
+      context('when MONGODB_LOG_MAX_DOCUMENT_LENGTH cannot be parsed as a uint', function () {
+        it('defaults to 1000', function () {
+          const loggerOptions = MongoLogger.resolveOptions(
+            {
+              MONGODB_LOG_MAX_DOCUMENT_LENGTH: 'invalid'
+            },
+            {}
+          );
+          expect(loggerOptions.maxDocumentLength).to.equal(1000);
+        });
+      });
+
+      context('when MONGODB_LOG_MAX_DOCUMENT_LENGTH can be parsed a uint', function () {
+        it('uses the passed value', function () {
+          const loggerOptions = MongoLogger.resolveOptions(
+            {
+              MONGODB_LOG_MAX_DOCUMENT_LENGTH: '500'
+            },
+            {}
+          );
+          expect(loggerOptions.maxDocumentLength).to.equal(500);
+        });
+      });
+    });
+
+    context('logDestination', function () {
```

Comment:
This looks great - I think we're missing a test for the condition where both MONGODB_LOG_PATH and mongodbLogPath are both non-empty strings, asserting that we preference the environment.